Sources/TensorFlow/Layer.swift

```diff
@@ -191,3 +191,35 @@ public final class Parameter<Scalar: TensorFlowScalar> {
         self.value = value
     }
 }
+
+/// A wrapper around a differentiable value with toggleable trainability.
+@_propertyWrapper
+public struct Trainable<Value: Differentiable> : Differentiable {
+    @noDerivative public var isTrainable: Bool = true
+    public var _value: Value
+
+    public init(initialValue: Value) {
+        _value = initialValue
+    }
+
+    @differentiable(vjp: _vjpValue)
+    public var value: Value {
+        get { _value.withoutDerivative() }
+        set { _value = newValue }
+    }
+
+    @usableFromInline
+    func _vjpValue() -> (value: Value, pullback: (Value.TangentVector) -> TangentVector) {
+        return (value, { [isTrainable = self.isTrainable] v in
+            if isTrainable {
```

Comment:
`isTrainable ? v : .zero`?

Reply:
I thought about ternary but chose not to originally. I'm *cool* with it though (`@Freezable` joke).